File: android/app/src/main/kotlin/de/tum/in/tumcampus/util/ColorExtension.kt
package de.tum.`in`.tumcampus.util

import android.graphics.Color

fun argbToColor(argb: Long): Int {
    val alpha = ((argb shr 24) and 0xFF) / 255f
    val red = ((argb shr 16) and 0xFF) / 255f
    val green = ((argb shr 8) and 0xFF) / 255f
    val blue = (argb and 0xFF) / 255f
    return Color.argb(alpha, red, green, blue)
}

File: ios/CalendarWidget/ColorExtension.swift
//
//  ColorExtension.swift
//  CalendarWidgetExtension
//
//  Created by Jakob Körber on 03.04.24.
//

import Foundation
import SwiftUI

extension Color {
    init(argb: UInt32) {
        let alpha = Double((argb >> 24) & 0xFF) / 255.0
        let red = Double((argb >> 16) & 0xFF) / 255.0
        let green = Double((argb >> 8) & 0xFF) / 255.0
        let blue = Double(argb & 0xFF) / 255.0
        self.init(red: red, green: green, blue: blue, opacity: alpha)
    }
}
